timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -0,0 +1,192 @@
+                1970-01-04       4
+                1970-01-05       5
+        >>> print(val_data)
+                                    target
+        item_id       timestamp
+        0_[None:None] 1970-01-01       1
+                      1970-01-02       2
+                      1970-01-03       3
+                      1970-01-04       4
+                      1970-01-05       5
+                      1970-01-06       6
+                      1970-01-07       7
+                      1970-01-08       8
+                      1970-01-09       9
+                      1970-01-10      10
+        0_[None:-2]   1970-01-01       1
+                      1970-01-02       2
+                      1970-01-03       3
+                      1970-01-04       4
+                      1970-01-05       5
+                      1970-01-06       6
+                      1970-01-07       7
+                      1970-01-08       8
+    """
+
+    def __init__(self, num_windows: int, overlap: int = 0):
+        self.num_windows = num_windows
+        self.overlap = overlap
+
+    def describe_validation_strategy(self, prediction_length):
+        return (
+            f"Will use the last {self.num_windows} windows (each with prediction_length {prediction_length} "
+            f"time steps and overlap {self.overlap}) as a hold-out validation set."
+        )
+
+    def _split(
+        self, ts_dataframe: TimeSeriesDataFrame, prediction_length: int
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        if self.overlap >= prediction_length:
+            raise ValueError(
+                f"SlidingWindowSplitter.overlap {self.overlap} must be < prediction_length {prediction_length}"
+            )
+        step_size = prediction_length - self.overlap
+
+        length_per_series = ts_dataframe.index.get_level_values(0).value_counts(sort=False)
+        num_total_validation_steps = prediction_length + step_size * (self.num_windows - 1)
+        num_too_short_series = (length_per_series <= num_total_validation_steps).sum()
+        if num_too_short_series > 0:
+            logger.warning(f"{num_too_short_series} are too short and won't appear in the training set")
+        if num_too_short_series == ts_dataframe.num_items:
+            raise ValueError(f"{self.name} produced an empty training set since all sequences are too short")
+
+        validation_dataframes = [append_suffix_to_item_id(ts_dataframe, "_[None:None]")]
+
+        for window_idx in range(1, self.num_windows):
+            ts_dataframe = ts_dataframe.slice_by_timestep(slice(None, -step_size))
+            total_offset = step_size * window_idx
+            next_val_dataframe = append_suffix_to_item_id(ts_dataframe, f"_[None:{-total_offset}]")
+            validation_dataframes.append(next_val_dataframe)
+
+        train_data = ts_dataframe.slice_by_timestep(slice(None, -prediction_length))
+
+        return train_data, pd.concat(validation_dataframes)
+
```

Comment:
At some point after benchmarking the current version I will implement an `AutomaticSplitter` that determines `num_windows` and `overlap` based on some heuristic.